src/classes/BDI_ObjectWrapper.cls

```diff
@@ -51,4 +51,19 @@ public with sharing class BDI_ObjectWrapper {
         this.fieldMappings = fieldMappings;
         this.dataImport = dataImport;
     }
+
+    public void setDataImportStatus(String status) {
+        dataImport.Status__c = status;
+    }
+
+    public void setImportedRecordStatusError(String errorMessage) {
+        String statusField = objMapping.Imported_Record_Status_Field_Name__c;
+        if (statusField != null && errorMessage != null) {
+            dataImport.put(statusField, errorMessage.left(255));
```

Comment:
Possibly consider truncating to 252 length and concatenating "..." just to make it clear that the message has been truncated?

Reply:
@ptewson-sfdo happy to do this as a future WI - currently the entire message is included in the Failure Information field which is a long text area so the User will at least be presented with the entire error message there if it is truncated here.